common/authorization/oauthAuthorizer.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "strings"
+    "time"
+
+    "github.com/cristalhq/jwt/v3"
+    "google.golang.org/grpc/metadata"
+
+    "github.com/uber/cadence/common/config"
+)
+
+type oauthAuthority struct {
+    cfg *config.Config
+}
+
+type jwtClaims struct {
+    name       string
+    permission string
+    domain     string
+    iat        int64
+}
+
+// NewOAuthAuthorizer creates a no-op authority
+func NewOAuthAuthorizer(
+    cfg *config.Config,
+) Authorizer {
+    return &oauthAuthority{
+        cfg: cfg,
+    }
+}
+
+func (a *oauthAuthority) Authorize(
+    ctx context.Context,
+    attributes *Attributes,
+) (Result, error) {
+
+    md, ok := metadata.FromIncomingContext(ctx)
```

Comment:
Cadence uses yrpc which is on top of both Thritf and gRPC.
See /common/rpc.go for how we define and use other headers.